aztec/src/main/kotlin/org/wordpress/aztec/source/Format.kt

```diff
@@ -6,22 +6,27 @@ import java.util.regex.Pattern
 object Format {
 
     // list of block elements
-    private val block = "div|span|br|blockquote|ul|ol"
+    private val block = "div|span|br|blockquote|ul|ol|li|h1|h2|h3|h4|h5|h6"
 
     fun addFormatting(content: String): String {
-        // let's use Jsoup for HTML formatting for now because it works out of the box
         val doc = Jsoup.parseBodyFragment(content)
-        return doc.body().html()
+
+        //remove newline around all non block elements
```

Comment:
I'm wary of altering regular expressions, especially if they already work.  We could nest the replacements like this below, but I'm not sure that's an improvement.  It actually might make the code more difficult to read.

```
return replaceAll(
    replaceAll(
        replaceAll(
            doc.body().html(),
            "(?<!</?$block>)\n<(/?(?!$block).)>",
            "<$1>"
        ),
        "<(/?(?!$block).)>\n(?!</?($block)>)",
        "<$1>"
    ),
    "(<br>)(?!\n)",
    "$1\n"
)
```

@0nko, would you like to take a look?